serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc

```diff
@@ -1 +1,159 @@
+
+Expressions are the mechanism defined by the link:{spec_doc_url}#workflow-expressions[Serverless Workflow Specification] for the states to interact with the data model. Kogito supports two expression languages: link:https://github.com/json-path/JsonPath[jsonpath] and link:https://stedolan.github.io/jq/manual/[jq]. The default one is `jq`, but you can change it to `jsonpath` by using the `expressionLang` property. This guide will focus on `jq` and discuss its usage in following uses cases: switch state conditions, action function arguments, and data filtering.
+
+
+== Switch Conditions
+
+Conditions inside a switch state allow the workflow designer to choose which path the flow should follow depending on the model content. A condition is an expression that returns a boolean when evaluated against the model. If the condition associated with a state transition returns true, that is the place for the flow to go. 
+
+For example, in link:{kogito_sw_examples_url}/serverless-workflow-greeting-quarkus[greetings] repository, we are selecting which message should be displayed to the user depending on his language of choice: English or Spanish. In computational terms, if the value of the property language is English, the constant literal to be injected on the property `message` will be __Hello from__… , else if the value of the same property is Spanish, then the injected `message` will be __Saludos desde…__.
+
+image::core/switch_condition.png[]
+
+The switch state contains this condition, which in turn contains two `jq` expression that returns a boolean. 
+
+[source,json]
+----
+"dataConditions": [
+        {
+          "condition": "${ .language == \"English\" }",
+          "transition": "GreetInEnglish"
+        },
+        {
+          "condition": "${ .language == \"Spanish\" }",
+          "transition": "GreetInSpanish"
+        }
+      ]
+----
+
+As the specification requires, all expressions are embedded within `${… }`. But Kogito is smart enough to infer that the string within the condition is an expression, so you can skip it and just write a shorter version of the previous code that behaves the same. 
+
+[source,json]
+----
+"dataConditions": [
+        {
+          "condition": ".language == \"English\"",
+          "transition": "GreetInEnglish"
+        },
+        {
+          "condition": ".language == \"Spanish\"",
+          "transition": "GreetInSpanish"
+        }
+      ]
+----
+
+== Function Arguments
+
+Serverless Workflow Specification allows defining link:{spec_doc_url}#workflow-functions[functions] that can be invoked several times by the workflow states. Every different function call might contain different arguments, which are specified using function arguments.
+
+A function definition can be found in the link:{kogito_sw_examples_url}/serverless-workflow-temperature-conversion[temperature conversion] repository. 
```

Comment:
I think this link should point the user to the sw.json file instead of to the project to free users to do an additional navigation to know where the referred functions is.
Additionally to the link you can also copy the function definition here in a source block, I think it can make users life easier.

Reply:
yes, this link is wrong, thanks